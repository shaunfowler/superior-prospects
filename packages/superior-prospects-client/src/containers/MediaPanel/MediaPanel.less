.media {
  li:hover {
    background-color: #f6f6f6;
    border-radius: 3px;
    transition: 300ms;
    cursor: pointer;
  }

  &__item {
    display: flex;
    align-items: center;
    padding: 5px 10px;

    .dropzone {
      margin-left: 16px;
      font-size: 12px;
      width: 100%;
    }

    &__avatar {
      margin-right: 10px;
    }

    &__text {
      align-self: center;
      height: 52px;
      display: flex;
      flex-direction: column;
      justify-content: center;
      font-size: 12px;
      word-break: break-all;

      h3,
      h4 {
        font-size: 12px;
        line-height: 14px;
        color: #333;
        margin-bottom: 5px;
        word-break: break-all;
      }

      p {
        font-size: 12px;
        line-height: 12px;
        color: #999;
      }
    }
  }
}
