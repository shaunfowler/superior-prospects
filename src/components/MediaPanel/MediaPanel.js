import React from "react";
import moment from "moment";
import Dropzone from "react-dropzone";
import ReactGA from "react-ga";
import fileTypeToIcon from "../../utils/fileTypeToIcon";
import {
  Avatar,
  List,
  ListItem,
  ListItemText,
  ListItemIcon,
  ListItemSecondaryAction,
  IconButton,
  Menu,
  MenuItem
} from "material-ui";
import {
  MoreVert as MoreVertIcon,
  Delete as DeleteIcon,
  FileUpload as FileUploadIcon
} from "material-ui-icons";
import { ListItemAvatar } from "material-ui";

const stripFileExtension = filename =>
  filename.substring(0, filename.lastIndexOf("."));

const formatMediaDate = date => moment(date).format("MMMM Do YYYY");

const trackMediaClick = fileName => {
  ReactGA.event({
    category: "Media",
    action: "Open media item",
    label: fileName
  });
};

const trackMediaMoreClick = () => {
  ReactGA.event({
    category: "Media",
    action: "Media 'more' button click"
  });
};

const trackDeleteMedia = () => {
  ReactGA.event({
    category: "Media",
    action: "Delete media item"
  });
};

class MediaPanel extends React.Component {
  state = {
    anchorEl: null
  };

  openMenu = event => {
    trackMediaMoreClick();
    this.setState({ anchorEl: event.currentTarget });
  };

  closeMenu = () => {
    this.setState({ anchorEl: null, selectedMediaId: null });
  };

  renderMenu = () => {
    const { deleteMedia } = this.props;
    const { selectedMediaId } = this.state;
    const { anchorEl } = this.state;
    return (
      <Menu
        anchorEl={this.state.anchorEl}
        open={Boolean(anchorEl)}
        onClose={this.closeMenu}
      >
        <MenuItem
          onClick={event => {
            this.closeMenu(event);
            deleteMedia(selectedMediaId);
            trackDeleteMedia();
          }}
        >
          <ListItemIcon>
            <DeleteIcon />
          </ListItemIcon>
          <ListItemText inset primary="Delete" />
        </MenuItem>
      </Menu>
    );
  };

  renderDropZone = () => {
    const { isAuthenticated, onFileDrop } = this.props;
    if (!isAuthenticated) {
      return null;
    }

    return (
      <ListItem disableGutters className="media__item">
        <Avatar>
          <FileUploadIcon />
        </Avatar>
        <Dropzone onDrop={onFileDrop} className="dropzone" multiple>
          <div className="name">Upload new items</div>
          <div className="date">
            <code>xlsx</code>, <code>docx</code>, <code>pdf</code>,{" "}
            <code>png</code>, <code>jpeg</code>
          </div>
        </Dropzone>
      </ListItem>
    );
  };

  renderMedia = () => {
    const { media } = this.props;
    if (!media) {
      return null;
    }

    return (
      <List>
        {this.renderDropZone()}
        {media.map(m => (
          <ListItem
            disableGutters
            key={m._id}
            className="media__item"
            href={`/api/static/${m.fileName}`}
            onClick={() => {
              trackMediaClick(m.fileName);
            }}
          >
            <ListItemAvatar>{fileTypeToIcon(m.fileName)}</ListItemAvatar>
            <ListItemText
              className="media__item__text"
              primary={stripFileExtension(m.fileName)}
              secondary={formatMediaDate(m.created)}
            />
            <ListItemSecondaryAction
              onClick={event => {
                this.setState({ selectedMediaId: m._id });
                this.openMenu(event);
              }}
            >
              <IconButton aria-label="More">
                <MoreVertIcon />
              </IconButton>
            </ListItemSecondaryAction>
          </ListItem>
        ))}
        {this.renderMenu()}
      </List>
    );
  };

  render() {
    return <div className="media">{this.renderMedia()}</div>;
  }
}

export default MediaPanel;
