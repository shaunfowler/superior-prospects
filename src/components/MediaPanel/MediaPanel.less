.media {
  &__item {
    display: flex;
    align-items: center;
    padding: 5px 10px;
    cursor: pointer;
    transition: 300ms;

    &:hover {
      background-color: #f6f6f6;
      border-radius: 3px;

      .name {
        color: #2196f3;
      }
    }

    &__avatar {
      margin-right: 10px;
    }

    &__text {
      align-self: center;
      height: 52px;
      display: flex;
      flex-direction: column;
      justify-content: center;

      .name {
        font-size: 12px;
        line-height: 12px;
        color: #333;
        margin-bottom: 4px;
        word-break: break-all;
      }

      .date {
        font-size: 12px;
        line-height: 12px;
        color: #999;
      }
    }
  }
}
