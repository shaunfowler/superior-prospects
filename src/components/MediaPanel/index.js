import MediaPanel from "./MediaPanel";
import "./MediaPanel.less";

export default MediaPanel;
