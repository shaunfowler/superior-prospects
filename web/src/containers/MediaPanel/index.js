import { connect } from "react-redux";
import { deleteMedia } from "../../redux/actions/mediaActions";
import MediaPanel from "./MediaPanel";
import "./MediaPanel.less";

export default connect(
  ({ user }) => ({
    isAuthenticated: user.isAuthenticated
  }),
  { deleteMedia }
)(MediaPanel);
