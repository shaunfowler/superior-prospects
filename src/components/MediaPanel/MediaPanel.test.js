import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import MediaPanel from "./";

describe("Properties container", () => {
  const media = [
    { _id: "1234", fileName: "test.png", created: "2018-04-16T12:00:00+00:00" }
  ];

  it("renders when not authenticated", () => {
    const props = {
      media,
      isAuthenticated: false
    };
    const output = shallow(<MediaPanel {...props} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it("renders when authenticated", () => {
    const props = {
      media,
      isAuthenticated: true
    };

    const output = shallow(<MediaPanel {...props} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
