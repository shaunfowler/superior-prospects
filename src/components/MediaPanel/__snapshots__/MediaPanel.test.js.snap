// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties container renders when authenticated 1`] = `
<div
  className="media"
>
  <div
    className="media__item"
  >
    <div
      className="media__item__avatar"
    >
      <WithStyles(Avatar)>
        <pure(FileUpload) />
      </WithStyles(Avatar)>
    </div>
    <div
      className="media__item__text"
    >
      <t
        className="dropzone"
        disableClick={false}
        disablePreview={false}
        disabled={false}
        maxSize={Infinity}
        minSize={0}
        multiple={true}
        preventDropOnDocument={true}
      >
        <div
          className="name"
        >
          Upload new items
        </div>
        <div
          className="date"
        >
          <code>
            xlsx
          </code>
          , 
          <code>
            docx
          </code>
          , 
          <code>
            pdf
          </code>
          ,
           
          <code>
            png
          </code>
          , 
          <code>
            jpeg
          </code>
        </div>
      </t>
    </div>
  </div>
  <div>
    <a
      href="/api/static/test.png"
      key="1234"
      target="_blank"
    >
      <div
        className="media__item"
      >
        <div
          className="media__item__avatar"
        >
          <WithStyles(Avatar)
            style={
              Object {
                "backgroundColor": "#ffcc80",
              }
            }
          >
            <pure(Photo) />
          </WithStyles(Avatar)>
        </div>
        <div
          className="media__item__text"
        >
          <div
            className="name"
          >
            test
          </div>
          <div
            className="date"
          >
            April 16th 2018
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;

exports[`Properties container renders when not authenticated 1`] = `
<div
  className="media"
>
  <div>
    <a
      href="/api/static/test.png"
      key="1234"
      target="_blank"
    >
      <div
        className="media__item"
      >
        <div
          className="media__item__avatar"
        >
          <WithStyles(Avatar)
            style={
              Object {
                "backgroundColor": "#ffcc80",
              }
            }
          >
            <pure(Photo) />
          </WithStyles(Avatar)>
        </div>
        <div
          className="media__item__text"
        >
          <div
            className="name"
          >
            test
          </div>
          <div
            className="date"
          >
            April 16th 2018
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;
